### ISSUE TITLE
Git repo with files encrypted by git-crypt in unlocked state causes error

### ISSUE THREAD

**Issue by PurpleBooth** [GoogleContainerTools/skaffold#623]
### Expected behavior

Skaffold run to correctly tag the docker container with the git commit.

### Actual behavior

You can find the source repo [here](https://github.com/PurpleBooth/FlattrackstatsBoutRSS)

Run 

```shell
$ git crypt unlock *.key
```

```shell
$ skaffold run
Starting build...
Sending build context to Docker daemon  53.41MB
Step 1/8 : FROM node:latest
 ---> f697cb5f31f8
Step 2/8 : RUN mkdir -p /usr/src/app
 ---> Using cache
 ---> 791ca4db4429
Step 3/8 : WORKDIR /usr/src/app
 ---> Using cache
 ---> 9bb160e8546f
Step 4/8 : COPY package.json /usr/src/app/
 ---> 12e3276ec621
Step 5/8 : RUN npm install
 ---> Running in da673c259b43
npm WARN deprecated libcipm@1.6.3: This version mistakenly dropped node@4 compatibility in a semver-patch release. Please stay on 1.6.2 or update to 2 or latest
npm WARN deprecated npm-registry-fetch@1.1.1: this version has a breaking change. use 1.1.0 or upgrade to latest
npm WARN deprecated socks@1.1.10: If using 2.x branch, please upgrade to at least 2.1.6 to avoid a serious bug with socket data flow and an import issue introduced in 2.1.0

> @newrelic/native-metrics@2.4.0 install /usr/src/app/node_modules/@newrelic/native-metrics
> node ./lib/pre-build.js install native_metrics

> /usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js clean configure
> /usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build -j 2 native_metrics
make: Entering directory '/usr/src/app/node_modules/@newrelic/native-metrics/build'
  CXX(target) Release/obj.target/native_metrics/src/native_metrics.o
  CXX(target) Release/obj.target/native_metrics/src/GCBinder.o
../src/GCBinder.cpp: In static member function 'static void nr::GCBinder::_doCallback(uv_work_t*, int)':
../src/GCBinder.cpp:51:5: warning: 'v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, v8::Local<v8::String>, int, v8::Local<v8::Value>*)' is deprecated (declared at ../../../nan/nan.h:950) [-Wdeprecated-declarations]
     );
     ^
  CXX(target) Release/obj.target/native_metrics/src/LoopChecker.o
  CXX(target) Release/obj.target/native_metrics/src/RUsageMeter.o
  SOLINK_MODULE(target) Release/obj.target/native_metrics.node
  COPY Release/native_metrics.node
make: Leaving directory '/usr/src/app/node_modules/@newrelic/native-metrics/build'
install successful: _newrelic_native_metrics-2_4_0-native_metrics-64-linux-x64
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN flattrackstats-bout-rss@0.1.3 No repository field.

added 843 packages in 34.722s
 ---> bf207055b89a
Step 6/8 : COPY . /usr/src/app
 ---> 41f6ba303209
Step 7/8 : EXPOSE 8081
 ---> Running in 3d9d1cd07214
 ---> a5491a8f6314
Step 8/8 : CMD [ "npm", "start" ]
 ---> Running in f7880ecfd7ca
 ---> 01e461fad830
Successfully built 01e461fad830
Successfully tagged b5836d64a179b90b95fde4ce38474061:latest
Error: build step: generating tag: opening git repo: 24:11: unquoted '\' must be followed by new line
```

### Information

- Skaffold version: version...

```shell
$ skaffold version
v0.6.1
```

- Operating system: ...

```
MacOS
```

- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v1alpha2
kind: Config
build:
  artifacts:
  - imageName: gcr.io/purplebooth-gke/github-purplebooth-flattrackstatsboutrss
deploy:
  helm:
    releases:
    - name: rollerrss
      chartPath: helm
      namespace: default
      values:
        image: gcr.io/purplebooth-gke/github-purplebooth-flattrackstatsboutrss
profiles:
  - name: gcb
    build:
      googleCloudBuild:
        projectId: purplebooth-gke
```

I feel this may in some way be related to my git config. You can find that [here](https://github.com/PurpleBooth/dotfiles/tree/master/git) if it's needed

### Steps to reproduce the behavior

Add [git-crypt](https://www.agwa.name/projects/git-crypt/) to some files in your repo (can be ignored with .dockerignore), and ensure it's unlocked

```shell
skaffold build
```

**Comment by dgageot** [comment]
Hi @PurpleBooth That's strange because I git cloned your repo and ran `skaffold build` without any issue. Your git global configuration shouldn't be a problem because the go library we use to interact with git repos simply ignores those files.

**Comment by PurpleBooth** [comment]
Ok! So I got it, it's to do with git-crypt. As soon as you unlock the repo it starts happening, lock it again it stops.

**Comment by PurpleBooth** [comment]
[Upstream bug](https://github.com/src-d/go-git/issues/681)

**Comment by dgageot** [comment]
@PurpleBooth Thank you for finding the root cause!

**Comment by dgageot** [comment]
Fixed with #639
